packages/babel-parser/src/parser/expression.js

```diff
@@ -2514,17 +2492,30 @@ export default class ExpressionParser extends LValParser {
     );
 
     this.next();
-    if (
-      this.match(tt.semi) ||
-      (!this.match(tt.star) && !this.state.type.startsExpr) ||
-      this.hasPrecedingLineBreak()
-    ) {
-      node.delegate = false;
-      node.argument = null;
-    } else {
-      node.delegate = this.eat(tt.star);
-      node.argument = this.parseMaybeAssign();
+    let delegating = false;
+    let argument = null;
+    if (!this.hasPrecedingLineBreak()) {
```

Comment:
I'm curious about why
```js
function* fn() {
  yield
  * []
}
```
is disallowed, it doesn't seem to be ambiguous :thinking:

Reply:
Good question. I have no idea why it is disallowed, maybe @bakkot can shred some light here?

Reply:
That decision was before my time, so I can only speculate. I agree it wouldn't break anything. My best guess is that it's future-proofing, to reserve the possibility of introducing `*` as a prefix operator.